### ISSUE TITLE
UtilityClassWithPublicConstructor should not be reported for sealed classes 

### ISSUE THREAD

**Issue by svick** [arturbosch/detekt#2005]
## Observed Behavior

I have a `sealed class` with no members of its own, except for a factory function:

```kotlin
sealed class Parent {
    companion object {
        fun create(foo: Int?, bar: String?): Parent? =
            when {
                foo != null -> FooChild(foo)
                bar != null -> BarChild(bar)
                else -> null
            }
    }
}

data class FooChild(val foo: Int) : Parent()

data class BarChild(val bar: String) : Parent()
```

This reports:

> UtilityClassWithPublicConstructor - [Parent]

## Expected Behavior

UtilityClassWithPublicConstructor should not be reported for `Parent`, or any other `sealed class`, since `sealed class`es don't have public constructors.

## Your Environment
* Version of detekt used: 1.1.0
